include/deal.II/lac/block_linear_operator.h

```diff
@@ -26,23 +26,365 @@
+  }
+
+  /**
+   * Create a BlockLinearOperator from a two-dimensional array @p ops of
+   * LinearOperator. This constructor calls the corresponding
+   * block_operator() specialization.
+   */
+  template<size_t m, size_t n>
+  BlockLinearOperator(const std::array<std::array<BlockType, n>, m> &ops)
+  {
+    *this = block_operator<m, n, Range, Domain>(ops);
+  }
+
+  /**
+   * Create a block-diagonal BlockLinearOperator from a one-dimensional
+   * array @p ops of LinearOperator. This constructor calls the
+   * corresponding block_operator() specialization.
+   */
+  template<size_t m>
+  BlockLinearOperator(const std::array<BlockType, m> &ops)
+  {
+    *this = block_diagonal_operator<m, Range, Domain>(ops);
+  }
+
+  /**
+   * Default copy assignment operator.
+   */
+  BlockLinearOperator<Range, Domain> &
+  operator=(const BlockLinearOperator<Range, Domain> &) = default;
+
+  /**
+   * Templated copy assignment operator for an object @p op for which the
+   * conversion function <code>block_operator</code> is defined.
+   */
+  template <typename Op>
+  BlockLinearOperator<Range, Domain> &operator=(const Op &op)
+  {
+    *this = block_operator<Range, Domain, Op>(op);
+    return *this;
+  }
+
+  /**
+   * Copy assignment from a two-dimensional array @p ops of LinearOperator.
+   * This assignment operator calls the corresponding block_operator()
+   * specialization.
+   */
+  template <size_t m, size_t n>
+  BlockLinearOperator<Range, Domain> &
+  operator=(const std::array<std::array<BlockType, n>, m> &ops)
+  {
+    *this = block_operator<m, n, Range, Domain>(ops);
+    return *this;
+  }
+
+  /**
+   * Copy assignment from a one-dimensional array @p ops of LinearOperator
+   * that creates a block-diagonal BlockLinearOperator.
+   * This assignment operator calls the corresponding block_operator()
+   * specialization.
+   */
+  template <size_t m>
+  BlockLinearOperator<Range, Domain> &
+  operator=(const std::array<BlockType, m> &ops)
+  {
+    *this = block_diagonal_operator<m, Range, Domain>(ops);
+    return *this;
+  }
+
+  /**
+   * TODO: Description
+   */
+  std::function<unsigned int()> n_block_rows;
+
+  /**
+   * TODO: Description
+   */
+  std::function<unsigned int()> n_block_cols;
+
+  /**
+   * TODO: Description
+   */
+  std::function<BlockType(unsigned int, unsigned int)> block;
+
+  //@}
+};
+
+namespace
+{
+  // Populate the LinearOperator interfaces with the help of the
+  // BlockLinearOperator functions
+  template <typename Range, typename Domain>
+  void populate_linear_operator_functions(BlockLinearOperator<Range, Domain> &op)
+  {
+    op.reinit_range_vector = [=](Range &v, bool fast)
+    {
+      const unsigned int m = op.n_block_rows();
+
+      // Reinitialize the block vector to m blocks:
+      v.reinit(m);
+
+      // And reinitialize every individual block with reinit_range_vectors:
+      for (size_t i = 0; i < m; ++i)
+        op.block(i, 0).reinit_range_vector(v.block(i), fast);
```

Comment:
Oh, yes, I think I recall now. We made matrices export their partitioning. It's all good, thanks for reminding me of that mechanism.